test/tla/TestGen.tla

```diff
@@ -0,0 +1,32 @@
+------------------------------ MODULE TestGen ---------------------------------
+(* A simple test for the data structure generator *)
+EXTENDS Integers, Apalache
+
+VARIABLES
+    (*
+     @typeAlias: MSG = [seqno: Int, ballot: Int];
+     @type: Set(MSG);
+    *)
+    msgs,
+    \* @type: Int;
+    seqno
+
+\* preparing a test, like you do it in unit tests
+Before ==
+    /\ seqno = Gen(1)
+    /\ msgs = Gen(5)
```

Comment:
What does `Gen(1) = Gen(1)` evaluate to?

Reply:
In the case of `seqno = Gen(1)`, the type of `Gen(1)` is `Int`, so you would get an integer constant. Since `Gen` has the polymorphic type `Int => a`, most likely, `Gen(1) = Gen(1)` will produces a type error in the error, as the model checker requires concrete types (monotypes).

Reply:
If you are wondering about what happens when one compares two values that are produced by a call to `Gen(1)`, there are two outcomes: Either they are equal, or they are not equal :-)

The purpose of `Gen` is to produce a logical term (a tree of constants) that can be evaluated by other expressions. It is up to the solver what solution to find. As in normal unit tests, you can restrict values when you are preparing the test.

Reply:
This operator is probably unusual for TLA+, as it does not preserve congruence, that is, two calls with the same argument may produce unequal results.

Reply:
Passing the desired type as an argument (and relying on the type checker) would remove all "unusualness".

Reply:
It will not. Two calls to `Gen(1, "Int")` would still be allowed to produce unequal values. I do not see see how passing the type as an argument would help.